{
  "schema_version": "1.4.0",
  "id": "GHSA-9frf-r7c7-j2vg",
  "modified": "2021-08-19T17:15:11Z",
  "published": "2021-08-25T20:54:14Z",
  "aliases": [
    "CVE-2021-29939"
  ],
  "summary": "Out of bounds write in stackvector",
  "details": "An issue was discovered in the stackvector crate before 1.0.9 for Rust. There is an out-of-bounds write in StackVec::extend if size_hint provides certain anomalous data.",
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:L"
    }
  ],
  "affected": [
    {
      "package": {
        "ecosystem": "crates.io",
        "name": "stackvector"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "1.0.9"
            }
          ]
        }
      ]
    }
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-29939"
    },
    {
      "type": "WEB",
      "url": "https://github.com/Alexhuszagh/rust-stackvector/issues/2"
    },
    {
      "type": "PACKAGE",
      "url": "https://github.com/Alexhuszagh/rust-stackvector"
    },
    {
      "type": "WEB",
      "url": "https://rustsec.org/advisories/RUSTSEC-2021-0048.html"
    }
  ],
  "database_specific": {
    "cwe_ids": [
      "CWE-787"
    ],
    "severity": "HIGH",
    "github_reviewed": true,
    "github_reviewed_at": "2021-08-19T17:15:11Z",
    "nvd_published_at": "2021-04-01T05:15:00Z"
  }
}